### PR TITLE
Adds extrinsic, decode/encode

### PR DESCRIPTION
**Pull request by justkawal**
## Any follow ups?
Please review PR: #122, before reviewing this PR.

## Description
* Decode Extrinsic from **RawBlock** and result in **DecodedBlockExtrinsic**
* Encode decoded-extrinsic to produce encoded raw-block.

## Testing Plan 
* Issue #130, to track implementation of extrinsic testing.

### CODE REVIEW

File: packages/substrate_metadata/lib/chain/chain.dart [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        getVersionDescription(blockNumber);
+
+    // Check if this is not empty, throw Exception if it is.
+    if (versionDescription == null) {
+      throw UnexpectedCaseException(
+          'Metadata not found for block: $blockNumber.');
+    }
+
+    final List<String> extrinsics =
+        decodedBlockExtrinsics.extrinsics.map((extrinsic) {
```
Comment (leonardocustodio) [inline packages/substrate_metadata/lib/chain/chain.dart:122]: Shouldn't be decodedBlock.extrinsics? Or just decodedBlockExtrinsics? I mean why two times extrinsics?

**Review by leonardocustodio — CHANGES_REQUESTED** [review]

File: packages/substrate_metadata/lib/chain/chain.dart (hunk 1 above)
Comment (justkawal) [reply packages/substrate_metadata/lib/chain/chain.dart:122]: `decodedBlockExtrinsics` has a property `extrinsics` which contains decoded extrinsics data, we're looping through that data and then encoding it back again as was previous in `rawBlock`.

File: packages/substrate_metadata/lib/chain/chain.dart [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    final VersionDescription? versionDescription =
+        getVersionDescription(rawBlock.blockNumber);
+
+    // Check if this is not empty, throw Exception if it is.
+    if (versionDescription == null) {
+      throw UnexpectedCaseException(
+          'Metadata not found for block: ${rawBlock.blockNumber}.');
+    }
+
+    final List<dynamic> extrinsics = rawBlock.extrinsics.map((hex) {
```
Comment (justkawal) [inline packages/substrate_metadata/lib/chain/chain.dart:84]: @leonardocustodio Here we're looping the extrinsics in `rawBlock` and decoding them and then saving them inside `DecodedBlockExtrinsic`.

**Review by leonardocustodio — APPROVED** [review]

**Review by leonardocustodio — APPROVED** [review]

**Comment by justkawal** [comment]
@gabrielokura @Lohann @pastre Can you please review it ?

File: packages/substrate_metadata/lib/chain_description/chain.dart [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import 'package:substrate_metadata/metadata_decoder.dart';
+import 'package:substrate_metadata/models/models.dart';
+import 'package:substrate_metadata/old/types_bundle.dart';
+import 'package:substrate_metadata/old/legacy_types_model.dart';
+import 'package:substrate_metadata/spec_version/spec_version.model.dart';
+import 'package:substrate_metadata/utils/spec_version_maker.dart';
+import 'package:polkadart_scale_codec/polkadart_scale_codec.dart'
+    as scale_codec;
+
+class Chain {
```
Comment (gabrielokura) [inline packages/substrate_metadata/lib/chain_description/chain.dart:14]: Why this class is duplicated?

Comment (justkawal) [reply packages/substrate_metadata/lib/chain_description/chain.dart]: Fixed ✅, Merging issue.

**Review by gabrielokura — APPROVED** [review]

**Pull request merged**